### ISSUE TITLE
linter: new rule no-invalid-selector

### ISSUE THREAD

**Issue by AdamJedl** [oxc-project/oxc#2711]
Create a new rule that show error when selector in document.querySelector() and document.querySelectorAll() is invalid.
For css parsing it would use [lightningcss](https://lightningcss.dev/)

Only reference I found about this is in eslint https://github.com/eslint/eslint/discussions/17125

I would like to work on this rule

**Comment by Boshen** [comment]
Yeah I disliked the need for a new eslint plugin when a rule will definitely catch bugs, which should be part of the core.

A PR is more than appreciated!

**Comment by Boshen** [comment]
Wait ... this is a little more complicated, it requires lightningcss, which would increase our binary size by a significant about (I don't know how much).

**Comment by AdamJedl** [comment]
@Boshen
I found out that only parcel_selectors and cssparser is needed which is a lot smaller.
```bash
cargo build -p oxc_cli --bin oxlint --release --target=x86_64-unknown-linux-gnu
```
Binary size:
- without no-invalid-selector: 8 197 288 B
- with no-invalid-selector: 8 508 584 B
+311 296 B +3.8%

I don't know if this is acceptable.

**Comment by Boshen** [comment]
Are there any more rules that utilizes css parsing? The cost would be justified if we have more supporting rules.

**Comment by Boshen** [comment]
Closing due to inactivity.
